File: app/schemas/watchlist.py
"""Watchlist/Favorites schemas."""

from datetime import datetime

from pydantic import BaseModel


class WatchlistItemResponse(BaseModel):
    """Response model for a watchlist item."""

    id: int
    media_item_id: int | None = None
    series_id: int | None = None
    added_at: datetime

    class Config:
        from_attributes = True


class WatchlistMediaResponse(BaseModel):
    """Response model for watchlist with media details."""

    id: int
    media_item_id: int | None = None
    series_id: int | None = None
    item_type: str  # "movie", "episode", or "series"
    added_at: datetime
    title: str
    poster_path: str | None
    media_type: str | None = None
    duration_seconds: int | None = None

    class Config:
        from_attributes = True


class WatchlistListResponse(BaseModel):
    """Response model for watchlist list."""

    items: list[WatchlistMediaResponse]
    total: int

File: frontend/src/lib/api/watchlist.ts
import { api } from "./client";

export interface WatchlistItem {
  id: number;
  media_item_id: number | null;
  series_id: number | null;
  item_type: string; // "movie", "episode", or "series"
  added_at: string;
  title: string;
  poster_path: string | null;
  media_type: string | null;
  duration_seconds: number | null;
}

export interface WatchlistResponse {
  items: WatchlistItem[];
  total: number;
}

export async function getWatchlist(): Promise<WatchlistResponse> {
  const response = await api.get<WatchlistResponse>("/api/v1/watchlist/");
  return response.data;
}

export async function addToWatchlist(mediaId: number): Promise<void> {
  await api.post(`/api/v1/watchlist/${mediaId}`);
}

export async function removeFromWatchlist(mediaId: number): Promise<void> {
  await api.delete(`/api/v1/watchlist/${mediaId}`);
}

export async function checkWatchlistStatus(mediaId: number): Promise<boolean> {
  const response = await api.get<{ in_watchlist: boolean }>(
    `/api/v1/watchlist/${mediaId}/status`,
  );
  return response.data.in_watchlist;
}

export async function addSeriesToWatchlist(seriesId: number): Promise<void> {
  await api.post(`/api/v1/watchlist/series/${seriesId}`);
}

export async function removeSeriesFromWatchlist(
  seriesId: number,
): Promise<void> {
  await api.delete(`/api/v1/watchlist/series/${seriesId}`);
}

export async function checkSeriesWatchlistStatus(
  seriesId: number,
): Promise<boolean> {
  const response = await api.get<{ in_watchlist: boolean }>(
    `/api/v1/watchlist/series/${seriesId}/status`,
  );
  return response.data.in_watchlist;
}
